### ISSUE TITLE
Fails on Windows

### ISSUE THREAD

**Issue by ShishkinDmitriy** [origin-energy/java-snapshot-testing#19]
Hi,

Start on Windows and get an error:
```
java.lang.IllegalArgumentException: character to be escaped is missing

	at java.util.regex.Matcher.appendReplacement(Matcher.java:809)
	at java.util.regex.Matcher.replaceAll(Matcher.java:955)
	at java.lang.String.replaceAll(String.java:2223)
	at au.com.origin.snapshots.SnapshotMatcher.start(SnapshotMatcher.java:35)
	at au.com.origin.snapshots.SnapshotMatcher.start(SnapshotMatcher.java:27)
	at au.com.origin.snapshots.junit5.SnapshotExtension.beforeAll(SnapshotExtension.java:20)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$7(ClassBasedTestDescriptor.java:359)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:359)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:189)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:78)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:132)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:248)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$5(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:226)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:199)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:132)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
	Suppressed: au.com.origin.snapshots.SnapshotMatchException: Could not find Snapshot Verifier for this thread
		at au.com.origin.snapshots.SnapshotMatcher.validateSnapshots(SnapshotMatcher.java:68)
		at au.com.origin.snapshots.junit5.SnapshotExtension.afterAll(SnapshotExtension.java:25)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeAfterAllCallbacks$13(ClassBasedTestDescriptor.java:425)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeAfterAllCallbacks$14(ClassBasedTestDescriptor.java:425)
		at java.util.ArrayList.forEach(ArrayList.java:1257)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeAfterAllCallbacks(ClassBasedTestDescriptor.java:425)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.after(ClassBasedTestDescriptor.java:213)
		at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.after(ClassBasedTestDescriptor.java:78)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:145)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:145)
		... 27 more

```

Looks like it tries to replace dots '.' to '\\' and backslash is a special symbol.

Reproduce on jshell:
```
C:\Program Files\java\openjdk11.0.5\bin>jshell.exe
|  Welcome to JShell -- Version 11.0.5
|  For an introduction type: /help intro

jshell> "asd.qwe.vxcv".replaceAll("\\.", File.separator);
|  Exception java.lang.IllegalArgumentException: character to be escaped is missing
|        at Matcher.appendExpandedReplacement (Matcher.java:1020)
|        at Matcher.appendReplacement (Matcher.java:998)
|        at Matcher.replaceAll (Matcher.java:1182)
|        at String.replaceAll (String.java:2126)
|        at (#1:1)
```

Possible fixes:
`String::replaceAll` -> `String::replace`:
```
jshell> "asd.qwe.vxcv".replace(".", File.separator);
$1 ==> "asd\\qwe\\vxcv"
```
Or wrap `File.separator` into `Matcher::quoteReplacement`:
```
jshell> "asd.qwe.vxcv".replaceAll("\\.", Matcher.quoteReplacement(File.separator));
$1 ==> "asd\\qwe\\vxcv"
```

Best regards,
Dmitry

**Comment by jackmatt2** [comment]
Thanks for the report

So this has only been tested under Mac/Linux currently.

If you can submit a PR we can review & release. I just don't have a Windows machine to test on currently.

**Comment by jackmatt2** [comment]
Released: 2.0.0-Beta5

**Comment by jackmatt2** [comment]
@ShishkinDmitriy have applied the fix you suggested - let me know if it is now fixed on windows.

**Comment by jackmatt2** [comment]
`2.0.0-Beta5` turned out to be a bad build and excluded a number of necessary dependencies. This should be fixed as of `2.0.0-Beta6` - maven sync can take some time.

**Comment by ShishkinDmitriy** [comment]
It is fixed
